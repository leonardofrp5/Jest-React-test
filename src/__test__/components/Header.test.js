import React from 'react';
import { mount, shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import Header from '../../components/Header';

it('Prueba de shallow', () => {
  const header = shallow(
    <ProviderMock>
      <Header />
    </ProviderMock>,
  );
  expect(header.length).toEqual(1);
});

it('Texto del Header', () => {
  const header = mount(
    <ProviderMock>
      <Header />
    </ProviderMock>,
  );
  expect(header.find('.Header-title').text()).toEqual('Platzi Store');
});

it('Test provider', () => {
  const header = create(
    <ProviderMock>
      <Header />
    </ProviderMock>,
  );
  expect(header.toJSON).toMatchSnapshot();
});
