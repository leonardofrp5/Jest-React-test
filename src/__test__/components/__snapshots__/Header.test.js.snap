// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test provider 1`] = `[Function]`;
